bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,556 @@
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    "BentoML currently not support {} repository".format(
+                        BentoUri.StorageType.Name(bento_pb.bento.uri.type)
+                    )
+                )
+            bento_path = bento_pb.bento.uri.uri
+
+            return self._add(deployment_pb, bento_pb, bento_path)
+        except BentoMLException as error:
+            deployment_pb.state.state = DeploymentState.ERROR
+            deployment_pb.state.error_message = f"Error: {str(error)}"
+            return ApplyDeploymentResponse(
+                status=error.status_proto, deployment=deployment_pb
+            )
+
+    def _add(self, deployment_pb, bento_pb, bento_path):
+        try:
+            if loader._is_remote_path(bento_path):
+                with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                    return self._add(deployment_pb, bento_pb, local_path)
+
+            deployment_spec = deployment_pb.spec
+            aws_ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+
+            artifact_s3_bucket_name = generate_aws_compatible_string(
+                "btml-{namespace}-{name}-{random_string}".format(
+                    namespace=deployment_pb.namespace,
+                    name=deployment_pb.name,
+                    random_string=uuid4().hex[:6].lower(),
+                )
+            )
+            create_s3_bucket_if_not_exists(
+                artifact_s3_bucket_name, aws_ec2_deployment_config.region
+            )
+            self.deploy_service(
+                deployment_pb,
+                deployment_spec,
+                bento_path,
+                aws_ec2_deployment_config,
+                artifact_s3_bucket_name,
+                aws_ec2_deployment_config.region,
+            )
+        except BentoMLException as error:
+            if artifact_s3_bucket_name and aws_ec2_deployment_config.region:
+                cleanup_s3_bucket_if_exist(
+                    artifact_s3_bucket_name, aws_ec2_deployment_config.region
+                )
+            raise error
+        return ApplyDeploymentResponse(status=Status.OK(), deployment=deployment_pb)
+
+    def delete(self, deployment_pb):
+        try:
+            deployment_spec = deployment_pb.spec
+            ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+            ec2_deployment_config.region = (
+                ec2_deployment_config.region or get_default_aws_region()
+            )
+            if not ec2_deployment_config.region:
+                raise InvalidArgument("AWS region is missing")
+
+            ecr_client = boto3.client("ecr", ec2_deployment_config.region)
+            cf_client = boto3.client("cloudformation", ec2_deployment_config.region)
```

Comment:
We should also remove the S3 bucket as well. Similar to the Lambda operator: https://github.com/bentoml/BentoML/blob/92adf34e42673c377a7087ec069b430a623cec29/bentoml/yatai/deployment/aws_lambda/operator.py#L450-L456

Reply:
fixed